sentry/src/main/java/io/sentry/EventProcessor.java

```diff
@@ -1,9 +1,15 @@
 package io.sentry;
 
+import io.sentry.protocol.SentryTransaction;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 public interface EventProcessor {
   @Nullable
   SentryEvent process(@NotNull SentryEvent event, @Nullable Object hint);
+
+  @Nullable
```

Comment:
I think this interface can be improvemed. Either we can also make `process` for `SentryEvent` return `event` by default, leaving user the choice which methods should be implemented, or make `EventProcessor` a parameterized type `T extends SentryBaseEvent` - this API would probably be the cleanest.

Reply:
`T extends SentryBaseEvent` was the first option, but it was a breaking change.
making `SentryEvent` also default would make sense though if a processor is only about transactions, makes sense

Reply:
made `process(SentryEvent event, @Nullable Object hint)` also a default method